framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1242,15 +1242,19 @@ public AnnotationMirror createIntRangeAnnotation(long from, long to) {
     }
 
     /**
-     * Create an {@code @IntRange} annotation from the range. May return BOTTOMVAL or UNKNOWNVAL.
+     * Create an {@code @IntRange} or {@code @IntVal} annotation from the range. May return
+     * BOTTOMVAL or UNKNOWNVAL.
      */
     public AnnotationMirror createIntRangeAnnotation(Range range) {
         if (range.isNothing()) {
             return BOTTOMVAL;
         } else if (range.isEverything()) {
             return UNKNOWNVAL;
-        } else {
+        } else if (range.isWiderThan(MAX_VALUES)) {
             return createIntRangeAnnotation(range.from, range.to);
+        } else {
```

Comment:
@mernst and I had a discussion regarding this before -- do we convert `@IntRange` to `@IntVal` when the range is not wider than 10?. I remember Mike's concern was that if we do this conversion, users might be confused if they are intended to use `@IntRange` but it get converted to `@IntVal` easily during constant analysis. For example:
```java
void test(@IntRange(from = 0, to = 1) a) {
  int b = a + 1; // b has @IntVal({1, 2}) instead of @IntRange(from = 1, to = 2);
}
```
So at that time, we decided not to make this conversion for now and would revisit this if the precision enhancement provided by the conversion is really needed.

If we do the conversion here, for consistency, I think we might also need to add the same logic in `ValueAnnotatedTypeFactory#ValueTypeAnnotator#replaceWithNewAnnoInSpecialCases`. And maybe add a warning that `@IntRange` is converted to `@IntVal` for better precision?